pkg/scale/bitvec.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale
+
+const (
+	// maxLen equivalent of `ARCH32BIT_BITSLICE_MAX_BITS` in parity-scale-codec
+	maxLen = 268435455
+	// byteSize is the number of bits in a byte
+	byteSize = 8
+)
+
+// BitVec is the implementation of the bit vector
+type BitVec struct {
+	size uint   `scale:"1"`
+	bits []bool `scale:"2"`
```

Comment:
I would assume you're adding the scale struct tags given that you need to length encode the size of this.  I don't think is required.  After reading the `parity-scale-codec` encoding of `BitVec` ([link](https://github.com/paritytech/parity-scale-codec/blob/372d6177d8e66cdfe0968c200c8e29fda674bf14/src/bit_vec.rs)):

```
impl<O: BitOrder, T: BitStore + Encode> Encode for BitSlice<T, O> {
	fn encode_to<W: Output + ?Sized>(&self, dest: &mut W) {
		let bits = self.len();
		assert!(
			bits <= ARCH32BIT_BITSLICE_MAX_BITS,
			"Attempted to encode a BitSlice with too many bits.",
		);
		Compact(bits as u32).encode_to(dest);

		// Iterate over chunks
		for chunk in self.chunks(core::mem::size_of::<T>() * 8) {
			let mut element = T::ZERO;
			element.view_bits_mut::<O>()[..chunk.len()].copy_from_bitslice(chunk);
			element.encode_to(dest);
		}
	}
}

impl<O: BitOrder, T: BitStore + Encode> Encode for BitVec<T, O> {
	fn encode_to<W: Output + ?Sized>(&self, dest: &mut W) {
		self.as_bitslice().encode_to(dest)
	}
}
```

It looks like the encoding of a `BitVec` is essentially the encoding of the underlying `BitSlice`, where the encoding is essentially a length encoded (ie. `Compact(u32)`) slice of a certain generic type `T`.  This is directly analogous to the scale encoding of a slice of a given type `T`, so`[]T` in Go. 

So whether or not you actually need to support `BitVec` in our interpretation needs to be researched.  If the use case from the Parity devs is to just get the most memory efficient implementation of describing a `Vec<u32>` on their side, it doesn't mean we need to re-implement `BitVec` and `BitSlice` in Go to ensure we can interoperate with the runtime.  Maybe it's just easier to map it as `[]T` in our use cases knowing that it's not the most memory efficient representation at this time.

Feel free to reach out and we can sync up again if any of this is unclear.

Reply:
I agree. Thank you!

Reply:
I've removed the `size`